src/Data/Audience.elm

```diff
@@ -36,6 +40,45 @@ type alias Audience =
 
 
 
+-- Decoders
+
+
+decoder : Decoder Audience
+decoder =
+    -- I know about Json.Decode.Pipeline but for small projects, I prefer to
+    -- limit the number of dependencies. Plus, using `mapX` leads to easier
+    -- to understand error messages than the applicative version.
```

Comment:
Good to see you know about the applicative pattern even if you chose not to use it here. (By the way, in our main codebase we've made the choice to do all decoders via `Decode.succeed Record |> Decode.andMap ... |> Decode.andMap ...` even if they fit into the `mapN` limit - just so that the codebase is consistent and any irregularities are very visible if the API is weird in some way.)

Can you expand on the difference in error messages between applicative and `mapX`? That's interesting to hear. Perhaps one shows "number of arguments" error and the other one "wanted to get `X`, but got `Y -> X`" or something similar? :eyes:

Reply:
Yeah, that is mainly because of the "wanted `X` but got `Y -> X`" which are quite cryptic. 

E.g, if you add a field to a record `MyRecord` and you have some decoder:
```
myRecordDecoder = 
   Decode.succeed MyRecord
      |> ....
```
you'll got this hard to understand error. But if you had:
```
myRecordDecoder =
    Decode.map3 MyRecord
     |> ...
```
the compiler can immediately see that `map3` and `MyRecord` don't match. But yeah, it is a matter of taste :) 

(and in production apps, I tend to use `Json.Decode.Pipeline`)

Reply:
Related: I've recently discovered that "uniformity" is not such an important goal and can easily lead to "rigidity", which is clearly not desirable. I did a long time deeply understand what Evan meant with this: https://guide.elm-lang.org/webapps/modules.html#growing-modules .